File: src/com/company/PriorityQ.java
package com.company;
//Приоритетная очередь, реализованная по возрастанию приоритетов
public class PriorityQ {
    private int maxSize;
    private int[] prArray;
    private int nItems;

    public PriorityQ(int size) {
        maxSize = size;
        nItems = 0;           //количество элементов
        prArray = new int[maxSize];
    }

    public void insert(int value)
    {
        int j;

        if (nItems == 0)
            prArray[nItems++] = value;
        else
        {
            for(j = nItems - 1; j >= 0; j--)
            {
                if (value > prArray[j])
                    prArray[j+1] = prArray[j];
                else
                    break;
            }
            prArray[j+1] = value;
            nItems++;
        }
    }

    public void display()
    {
        for (int i = 0; i < nItems; i++)
        {
            System.out.print(prArray[i] +  " ");
        }
    }

    public int remove()   //Извлечение минимального элемента
    {
        return prArray[--nItems];
    }

    public int peelMin()
    {
        return prArray[nItems - 1];
    }

    public boolean isEmpty()
    {
        return (nItems == 0);
    }

    public boolean isFull()
    {
        return (nItems == maxSize);
    }


}

File: StructuresOnKotlin/src/priorityQ.kt


class priorityQ(private val size:Int) {


    private val queArray = LongArray(size)
    private var nElms: Int = 0

    fun isEmpty(): Boolean = nElms == 0
    fun isFull(): Boolean = nElms == size

    fun peek():Long = queArray[nElms-1]

    fun insert(a: Long) {
        if(nElms == 0) {
            queArray[nElms++] = a
        }
        else
        {
            if (isFull())
                throw Exception("Массив полон.Пиздец:((")
            else
            {
                var temp = nElms - 1
                while (a > queArray[temp])
                {
                    queArray[temp+1] = queArray[temp]
                    temp--
                    if (temp == -1) //Проверяем, дошел ли перебор до конца очереди
                    {
                        break
                    }
                }
            queArray[temp+1] = a
            nElms++
            }
        }
    }

    fun displayQueue(){
        for (i in nElms until 0)
        {
            print(" ${queArray[i]} ")
        }
        println()
    }

    fun remove():Long
    {
        if (isEmpty())
            throw Exception("Массив пуст и нечего ремувывать.")
        else
            return queArray[--nElms]
    }

    fun removeAll(){
        while (!isEmpty())
            println("${this.remove()}")
    }


}